evap/results/views.py

```diff
@@ -95,26 +94,24 @@ def course_detail(request, semester_id, course_id):
         questionnaire_result.question_results = filtered_question_results
 
     # remove empty questionnaire_results
-    questionnaire_results = [questionnaire_result for questionnaire_result in questionnaire_results if questionnaire_result.question_results]
+    for contribution_result in course_result.contribution_results:
+        contribution_result.questionnaire_results = [q for q in contribution_result.questionnaire_results if q.question_results]
```

Comment:
maybe make this a method? That would keep the verbosity while removing the comment and making this method shorter

Reply:
I think even more code in this method could be seperate methods in CourseResult:
- filter empty headings
- split into top, contributor and bottom

What do you think?

Reply:
that would work and it would make this method here shorter, but i'm not a big fan of methods that are called in exactly one place. `add_warnings` i put into a method because it was lots of code that didn't do very much. actually i'm thinking about moving the has_votes part back into this method...

I mean, of course we can put every section into a separate method, but what's the point? then everything is in a method, what gives? it gives me warm fuzzy feelings as well but i can't find a rational reasoning why that would help. maybe you can.

in the last fixup commit, i tried to condense the split part, but if anything it just got more horizontal. @He3lixxx could you tell me whether you would prefer that in case we don't use methods for the sections?

Reply:
I looked at it again and I think it doesn't really become better, so I agree with you and would leave it like that.

About the last thing, condensing the split part: I don't know to be honest. I don't think I've ever seen people use an if expression to determine the object they want to call a method on in-place, but I don't see any reason to not do it, so idk.